README.md

```diff
@@ -9,9 +9,15 @@ generic [Set](https://en.wikipedia.org/wiki/Set) implementations for Go.
 
 ---
 
-**PSA** October 2023 - The **v2** version of this package has been published, starting at
-tag version `v2.1.0`. A description of the changes including backwards incompatibilities
-can be found in https://github.com/hashicorp/go-set/issues/73
+**PSA** August 2024 - The **v3** version of this package has been published,
+starting at tag version `v3.0.0`. A description of the changes including
+backwards incompatibilities can be found in https://github.com/hashicorp/go-set/issues/90
```

Comment:
The `go.mod` sets a go version but should we note here as well that you need to be on a version of Go with the iterator package?

Reply:
good idea, added